jdaviz/core/freezable_state.py

```diff
@@ -53,6 +54,36 @@ def _reset_x_limits(self, *event):
             self.x_min = x_min
             self.x_max = x_max
 
+    def _convert_units_y_limits(self, old_unit, new_unit):
+        # override glue's _convert_units_y_limits to account
+        # for equivalencies.  This converts all four corners
+        # of the limits to set new limits that contain those
+        # same corners
+
+        if old_unit != new_unit:
+            if old_unit is None or new_unit is None:
+                self._reset_y_limits()
+                return
+            x_corners = np.array([self.x_min, self.x_min, self.x_max, self.x_max])
+            y_corners = np.array([self.y_min, self.y_max, self.y_min, self.y_max])
+            spectral_axis = x_corners * u.Unit(self.x_display_unit)
+
+            # NOTE: this uses the scale factor from the first found layer.  We may want to
+            # generalize this to iterate over all scale factors if/when we support multiple
+            # flux cubes (with potentially different pixel scale factors).
+            for layer in self.layers:
+                if psc := getattr(layer.layer, 'meta', {}).get('_pixel_scale_factor', None):  # noqa
+                    spectral_axis.info.meta = {'_pixel_scale_factor',
+                                               psc}
+                    break
+            else:
+                spectral_axis.info.meta = {}
+
+            y_corners_new = flux_conversion(y_corners, old_unit, new_unit, spectral_axis=spectral_axis)  # noqa
```

Comment:
will this handle equivalencies needed for surface brightness > surface brightness conversions, that may need pixel scale factor?

Reply:
yes, it should

Reply:
where does it get that from if there is no spectrum.meta or additional equivalency?

Reply:
the for-loop directly above this (it does rely on `layer.layer.meta`, but that should be set by spectral extraction)

Reply:
On L74 there is a loop to find the scale factor if it exists, and if so, to attach it to the metadata of the `spectral_axis`. Then there is a small modification to `flux_conversion` to account for passing the scale factor to `spectal_axis` instead of `spec`.

Reply:
thanks, i see what's happening now!